### ISSUE TITLE
The provided lr scheduler `StepLR` doesn't follow PyTorch's LRScheduler API

### ISSUE THREAD

**Issue by vors** [Lightning-AI/lightning#16211]
### Bug description

Hello! This is half bug report, half question.

First the question part: does tip of the lightning `master` tracks (==compatible) the latest stable release of PyTorch or the tip of PyTorch `master`? If the answer is "release" then you probably should disregard the bug report part :) 

When I run the PTL master (commit a808ff9b1c3e2e4a5953f8e6bf7d9525f0dc6cf4) against the PT master (commit 67c53d50e5b34a536cce2ad98b9ca5f65ff8a34d), I'm getting the following error (see logs section)

### How to reproduce the bug

```python
Sorry for not providing a clear repro, it involves some of our codebase that I cannot share. I hoping that even without the repro the error message is descriptive enough that it's useful.
```

### Error messages and logs

```
cruise_exedir/pytorch_lightning/trainer/trainer.py:774: in fit
    self._fit_impl, model, train_dataloaders, val_dataloaders, datamodule, ckpt_path
cruise_exedir/pytorch_lightning/trainer/trainer.py:726: in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
cruise_exedir/pytorch_lightning/trainer/trainer.py:814: in _fit_impl
    results = self._run(model, ckpt_path=self.ckpt_path)
cruise_exedir/pytorch_lightning/trainer/trainer.py:1220: in _run
    self.strategy.setup(self)
cruise_exedir/pytorch_lightning/strategies/ddp.py:179: in setup
    self.setup_optimizers(trainer)
cruise_exedir/pytorch_lightning/strategies/strategy.py:129: in setup_optimizers
    self.lightning_module
cruise_exedir/pytorch_lightning/core/optimizer.py:195: in _init_optimizers_and_lr_schedulers
    _validate_scheduler_api(lr_scheduler_configs, model)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

lr_scheduler_configs = [LRSchedulerConfig(scheduler=<torch.optim.lr_scheduler.StepLR object at 0x7f336aa07e80>, name=None, interval='epoch', frequency=1, reduce_on_plateau=False, monitor=None, strict=True, opt_idx=0)]
model = MockCruiseLightningModule(
  (model): MockModel(
    (mlp): Sequential(
      (0): Linear(in_features=128, out_features=64, bias=True)
      (1): ReLU()
    )
    (criteria): NLLLoss()
  )
)

    def _validate_scheduler_api(lr_scheduler_configs: List[LRSchedulerConfig], model: "pl.LightningModule") -> None:
        for config in lr_scheduler_configs:
            scheduler = config.scheduler
            if not isinstance(scheduler, _Stateful):
                raise TypeError(
                    f"The provided lr scheduler `{scheduler.__class__.__name__}` is invalid."
                    " It should have `state_dict` and `load_state_dict` methods defined."
                )
    
            if not isinstance(scheduler, LRSchedulerTypeTuple) and not is_overridden("lr_scheduler_step", model):
                raise MisconfigurationException(
>                   f"The provided lr scheduler `{scheduler.__class__.__name__}` doesn't follow PyTorch's LRScheduler"
                    " API. You should override the `LightningModule.lr_scheduler_step` hook with your own logic if"
                    " you are using a custom LR scheduler."
                )
E               pytorch_lightning.utilities.exceptions.MisconfigurationException: The provided lr scheduler `StepLR` doesn't follow PyTorch's LRScheduler API. You should override the `LightningModule.lr_scheduler_step` hook with your own logic if you are using a custom LR scheduler.

cruise_exedir/pytorch_lightning/core/optimizer.py:351: MisconfigurationException
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 1.10):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

**Comment by vors** [comment]
Sorry I actually run this on the old lightning, please disregard it.
